### ISSUE TITLE
Breaking change: Update FrameworkName to ".NET"

### ISSUE THREAD

**Issue by richlander** [dotnet/runtime#33680]
See: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs#L11

This [code](https://github.com/richlander/testapps/blob/master/versioninfo/Program.cs) produces this output today (for 5.0):

**.NET Core info**
Version: 5.0.0
FrameworkDescription: .NET Core 5.0.0-preview.1.20120.5
CoreCLR Build: 5.0.0-preview.1.20120.5
CoreCLR Hash: 3c523a6a7a3ebc25fe524359127b1d8846e23ea3
CoreFX Build: 5.0.0-preview.1.20120.5
CoreFX Hash: 3c523a6a7a3ebc25fe524359127b1d8846e23ea3

**Comment by danmosemsft** [comment]
@jkotas do you plan to update markdowns in this repo also?

**Comment by jkotas** [comment]
> @jkotas do you plan to update markdowns in this repo also?

I have done a pass over the repo docs in #33694

**Comment by jkotas** [comment]
I am sure there is more cleanup that can be done in the repo docs and comments, but I think we won't be able to fix all references to .NET Core. For example, if a comment talks about .NET Core 2.0, it should stay as .NET Core 2.0. Changing it to .NET 2.0 would be ambiguous (does it mean .NET Framework 2.0 or .NET Core 2.0?)

**Comment by danmosemsft** [comment]
@jkotas there are 244 hits for .NET Core in docs/**md. I take your point about ambiguity with .NET Framework. But if a year or two from now someone comes fresh to this repo, and is not interested in history, should they have to know about .NET Core? 

I don't think we do - but I am also not sure how to reword text like "all .NET Core assemblies are strong-named". "all .NET Core and .NET 5+ assemblies are strong-named" is super clumsy. Do we just wait a couple years for ambiguity to reduce then make it "all .NET assemblies are strong-named"

**Comment by danmosemsft** [comment]
I don't mind making the md exits BTW, if it's clear how to do them.

**Comment by jkotas** [comment]
> if a year or two from now someone comes fresh to this repo, and wants to find out about .NET 5+

I agree that the need to learn about .NET Core should and will be fading away over time, but .NET Core is always going to be part of the history and it won't be ever possible to erase it completely. For example, I expect that it is always going to be in the repo glossary.

> all .NET Core assemblies are strong-named

I would reword this one as "all .NET Runtime assemblies are strong named".

> .NET Core host probing

I think this one would be fine to change to ".NET host probing". There is very low risk of ambiguity with .NET Framework.

> ECMA 372 is supported by the .NET Framework, but not .NET Core

This should say "ECMA 372 is supported by the .NET Runtime on Windows platforms only."

Other examples:

> For .NET Core version 1.x and 2.0.x

This should either stay as is; or the whole paragraph should be deleted if it is no longer relevant.

> select `.NET Core` from the selection dropdown

This should stay as is until the UX is changed.

**Comment by danmosemsft** [comment]
Can you clarify what you mean by ".NET Runtime" in your examples? I assume you don't mean it to include .NET Framework. Does it mean ".NET Core and .NET 5+" ? (I do not know where we support ECMA372)

**Comment by jkotas** [comment]
I think ".NET runtime" is a good name to use for a general reference to .NET Framework or .NET Core. We have used this name in .NET Framework days too.

> where we support ECMA372

.NET Framework 2.0+, .NET Core 3.0+, .NET 5+

**Comment by danmosemsft** [comment]
Seems enough was done here cc @richla

**Comment by richlander** [comment]
Sounds good. Thanks for all the work, Jan! Looks great.

**Comment by ericstj** [comment]
@ViktorHofer do we have a breaking change notice for the TFM change?  If not can we add one here: https://github.com/dotnet/docs/issues/new?template=dotnet-breaking-change.md

**Comment by ViktorHofer** [comment]
We don't have one yet AFAIK. What would be the breaking change here? That we rename the short target framework moniker from netcoreapp to net? cc @terrajobst

**Comment by danmosemsft** [comment]
Could break anyone testing Framework name (maybe various API) eg https://github.com/gfoidl/Stochastics/issues/88

**Comment by ViktorHofer** [comment]
Sorry I was reasoning to Eric's question. I understand that renaming the FrameworkName is a breaking change. I guess even for the TargetFrameworkMoniker, which shouldn't be read/persisted anywhere as it's inferred properties should be used, a breaking change notice makes sense.

**Comment by ericstj** [comment]
We can enumerate the places the new moniker may appear.  The thing @danmosemsft mentioned actually doesn't change.    We left it at .NETCoreApp:
```c#
[assembly: TargetFramework(".NETCoreApp,Version=v5.0", FrameworkDisplayName = "")]
```
The same seems to be true for AppDomainSetup.TargetFrameworkName.

@Jkotas changed what RuntimeInformation returns here: 
https://github.com/dotnet/runtime/commit/410b992bfba530841451fc4e1857be8ea1a8f92a

Perhaps that's the only place?  It might be worthwhile to just talk in general about the change and how it works and how it is largely non-breaking.

**Comment by ViktorHofer** [comment]
Filed the breaking change issue: https://github.com/dotnet/docs/issues/20958

**Comment by ericstj** [comment]
Thanks @ViktorHofer, removed the label.
